horovod/common/operations.cc

```diff
@@ -1167,8 +1170,20 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
   // We will ask for multiple threads, so other libraries like mpi4py can
   // be used together with Horovod if multi-threaded MPI is installed.
   int provided;
-  MPI_Init_thread(NULL, NULL, MPI_THREAD_MULTIPLE, &provided);
 
+  // Calls MPI_Init if horovod_global.auto_mpi_init is set otherwise prints warning
+  if(horovod_global.auto_mpi_init){
+    int is_mpi_initialized= 0;
+    MPI_Initialized(&is_mpi_initialized);
+    if(!is_mpi_initialized)
```

Comment:
Let's make logic simpler - if `mpi_init=True`, let's just add a: WARNING: MPI has already been initialized, please use `hvd.init(mpi_init=False, mpi_finalize=False)` if you're using other frameworks that initialize MPI prior to Horovod.